FWCore/SharedMemory/interface/ControllerChannel.h

```diff
@@ -0,0 +1,132 @@
+#ifndef FWCore_SharedMemory_ControllerChannel_h
+#define FWCore_SharedMemory_ControllerChannel_h
+// -*- C++ -*-
+//
+// Package:     FWCore/SharedMemory
+// Class  :     ControllerChannel
+//
+/**\class ControllerChannel ControllerChannel.h " FWCore/SharedMemory/interface/ControllerChannel.h"
+
+ Description: Primary communication channel for the Controller process
+
+ Usage:
+    Works in conjunction with the WorkerChannel
+
+*/
+//
+// Original Author:  Chris Jones
+//         Created:  21/01/2020
+//
+
+// system include files
+#include <string>
+#include <iostream>
+#include "boost/interprocess/managed_shared_memory.hpp"
+#include "boost/interprocess/sync/named_mutex.hpp"
+#include "boost/interprocess/sync/named_condition.hpp"
+#include "boost/interprocess/sync/scoped_lock.hpp"
+
+// user include files
+#include "FWCore/Utilities/interface/Transition.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+// forward declarations
+
+namespace edm::shared_memory {
+  class ControllerChannel {
+  public:
+    /** iName is used as the base for the shared memory name. The full name uses iID as well as getpid() to create the value sharedMemoryName().
+     iID allows multiple ControllChannels to use the same base name iName.
+     */
+    ControllerChannel(std::string const& iName, int iID);
+    ~ControllerChannel();
+    ControllerChannel(const ControllerChannel&) = delete;
+    const ControllerChannel& operator=(const ControllerChannel&) = delete;
```

Comment:
Can `ControllerChannel` object be moved? Either way, I'd suggest to add explicit move operations for clarity.

Reply:
done